components/sup/src/manager/service/mod.rs

```diff
@@ -120,6 +120,32 @@ enum BindStatus<'a> {
     Unknown(Error),
 }
 
+/// Encapsulate changes to /hooks or /config.
+#[derive(Default)]
+struct TemplateUpdate {
+    hooks:                 HookCompileTable,
+    config_changed:        bool,
+    have_reconfigure_hook: bool,
+}
+
+impl TemplateUpdate {
+    fn new(hooks: HookCompileTable, config_changed: bool, have_reconfigure_hook: bool) -> Self {
+        Self { hooks,
+               config_changed,
+               have_reconfigure_hook }
+    }
+
+    fn needs_restart(&self) -> bool {
```

Comment:
Some documentation comments about _why_ the logic is this way (on this as well as `needs_reconfigure` below) would be useful. 

Having the reason why `run` and `post_run` are the "special" hooks clearly spelled out will also be a useful guide to Future Us as we add additional hooks.